src/promptflow/promptflow/core/_flow.py

```diff
@@ -0,0 +1,419 @@
+        # will generate same name if connection has same content
+        connection_dict = connection._to_dict()
+        connection_name = f"connection_{hash(json.dumps(connection_dict, sort_keys=True))}"
+        return connection_name
+
+    def _to_dict(self):
+        return {
+            "connections": self.connections,
+            "variant": self.variant,
+            "overrides": self.overrides,
+            "streaming": self.streaming,
+        }
+
+    def __eq__(self, other):
+        if isinstance(other, FlowContext):
+            return self._to_dict() == other._to_dict()
+        return False
+
+    def __hash__(self):
+        self._resolve_connections()
+        return hash(json.dumps(self._to_dict(), sort_keys=True))
+
+
+class FlowBase(abc.ABC):
+    def __init__(self, *, data: dict, code: Path, path: Path, **kwargs):
+        self._context = FlowContext()
+        # flow.dag.yaml's content if provided
+        self._data = data
+        # working directory of the flow
+        self._code = Path(code).resolve()
+        # flow file path, can be script file or flow definition YAML file
+        self._path = Path(path).resolve()
+        # hash of flow's entry file, used to skip invoke if entry file is not changed
+        self._content_hash = kwargs.pop("content_hash", None)
+        super().__init__(**kwargs)
+
+    @property
+    def context(self) -> FlowContext:
+        return self._context
+
+    @context.setter
+    def context(self, val):
+        if not isinstance(val, FlowContext):
+            raise UserErrorException("context must be a FlowContext object, got {type(val)} instead.")
+        self._context = val
+
+    @property
+    def code(self) -> Path:
+        """Working directory of the flow."""
+        return self._code
+
+    @property
+    def path(self) -> Path:
+        """Flow file path. Can be script file or flow definition YAML file."""
+        return self._path
+
+    @classmethod
+    # pylint: disable=unused-argument
+    def _resolve_cls_and_type(cls, data, params_override):
+        """Resolve the class to use for deserializing the data. Return current class if no override is provided.
+        :param data: Data to deserialize.
+        :type data: dict
+        :param params_override: Parameters to override, defaults to None
+        :type params_override: typing.Optional[list]
+        :return: Class to use for deserializing the data & its "type". Type will be None if no override is provided.
+        :rtype: tuple[class, typing.Optional[str]]
+        """
+        return cls, "flow"
+
+
+class Flow(FlowBase):
+    """A Flow in the context of PromptFlow is a sequence of steps that define a task.
+    Each step in the flow could be a prompt that is sent to a language model, or simply a function task,
+    and the output of one step can be used as the input to the next.
+    Flows can be used to build complex applications with language models.
+
+    Simple Example:
```

Comment:
copy this: 
![image](https://github.com/microsoft/promptflow/assets/47586720/e0166b7e-5f77-4cce-912a-a7a69abe3da1)

https://microsoft.github.io/promptflow/reference/python-library-reference/promptflow.tracing.html#promptflow.tracing.trace

Reply:
![Uploading image.png…]()

Reply:
![image](https://github.com/microsoft/promptflow/assets/2208599/701075c2-b0f5-4319-acb4-472d28ad2cef)